test/spec/modules/smaatoBidAdapter_spec.js

```diff
@@ -368,222 +102,537 @@ describe('smaatoBidAdapterTest', () => {
+              return adType;
+            }
+          }
+        }
+      };
+    };
+
+    it('returns empty array on no bid responses', () => {
+      const response_with_empty_body = { body: {} }
+
+      const bids = spec.interpretResponse(response_with_empty_body, request);
+
+      expect(bids).to.be.empty
     });
-  });
 
-  describe('interpretResponse', () => {
     it('single image reponse', () => {
-      const bids = spec.interpretResponse(openRtbBidResponse(ADTYPE_IMG), request);
-      assert.deepStrictEqual(bids, interpretedBidsImg);
+      const bids = spec.interpretResponse(buildOpenRtbBidResponse(ADTYPE_IMG), request);
+
+      expect(bids).to.deep.equal([
+        {
+          requestId: '226416e6e6bf41',
+          cpm: 0.01,
+          width: 350,
+          height: 50,
+          ad: '<div style="cursor:pointer" onclick="fetch(decodeURIComponent(\'https%3A%2F%2Fprebid%2Ftrack%2Fclick%2F1\'), {cache: \'no-cache\'});;window.open(decodeURIComponent(\'https%3A%2F%2Fprebid%2Ftrack%2Fctaurl\'));"><img src="https://prebid/static/ad.jpg" width="320" height="50"/><img src="https://prebid/track/imp/1" alt="" width="0" height="0"/><img src="https://prebid/track/imp/2" alt="" width="0" height="0"/></div>',
+          ttl: 300,
+          creativeId: 'CR69381',
+          dealId: '12345',
+          netRevenue: true,
+          currency: 'USD',
+          meta: {
+            advertiserDomains: ['smaato.com'],
+            agencyId: 'CM6523',
+            networkName: 'smaato',
+            mediaType: 'banner'
+          }
+        }
+      ]);
     });
+
     it('single richmedia reponse', () => {
-      const bids = spec.interpretResponse(openRtbBidResponse(ADTYPE_RICHMEDIA), request);
-      assert.deepStrictEqual(bids, interpretedBidsRichmedia);
+      const bids = spec.interpretResponse(buildOpenRtbBidResponse(ADTYPE_RICHMEDIA), request);
+
+      expect(bids).to.deep.equal([
+        {
+          requestId: '226416e6e6bf41',
+          cpm: 0.01,
+          width: 350,
+          height: 50,
+          ad: '<div onclick="fetch(decodeURIComponent(\'https%3A%2F%2Fprebid%2Ftrack%2Fclick%2F1\'), {cache: \'no-cache\'});"><div><h3>RICHMEDIA CONTENT</h3></div><img src="https://prebid/track/imp/1" alt="" width="0" height="0"/><img src="https://prebid/track/imp/2" alt="" width="0" height="0"/></div>',
+          ttl: 300,
+          creativeId: 'CR69381',
+          dealId: '12345',
+          netRevenue: true,
+          currency: 'USD',
+          meta: {
+            advertiserDomains: ['smaato.com'],
+            agencyId: 'CM6523',
+            networkName: 'smaato',
+            mediaType: 'banner'
+          }
+        }
+      ]);
     });
+
     it('single video reponse', () => {
-      const bids = spec.interpretResponse(openRtbBidResponse(ADTYPE_VIDEO), request);
-      assert.deepStrictEqual(bids, interpretedBidsVideo);
+      const bids = spec.interpretResponse(buildOpenRtbBidResponse(ADTYPE_VIDEO), request);
```

Comment:
I'd say we need a case for multi-imp request here as well, just to know what is returned

Reply:
This is for interpreting the response from our server though :confused:

Reply:
ah okay